src/main/java/com/mroxny/mtba/MultiplexManager.java

```diff
@@ -0,0 +1,196 @@
+     * @param endTime maximum movie start time
+     * @return a list of screenings in a given time period
+     */
+    public List<Screening> listScreenings( LocalDate day,  LocalTime startTime, LocalTime endTime) {
+        return screenings.stream()
+                .filter(screening -> screening.getDay().isEqual(day) &&
+                        screening.getStartTime().isAfter(startTime) &&
+                        screening.getStartTime().isBefore(endTime))
+                .sorted(Comparator.comparing((Screening screening) -> screening.getMovie().getTitle())
+                        .thenComparing(Screening::getStartTime))
+                .collect(Collectors.toList());
+    }
+
+    public Screening getScreening(int screeningId) {
+
+        for (Screening s : screenings) {
+            if (s.getScreeningId() == screeningId) return s;
+        }
+        return null;
+    }
+
+    public Room getRoom(int roomNumber){
+        for (Room r : rooms) {
+            if (r.getRoomNumber() == roomNumber) return r;
+        }
+        return null;
+    }
+
+    public List<Screening> listAllScreenings(){
+        return screenings;
+    }
+    public Reservation getReservation(int id){
+        for(Reservation r : reservations)
+            if(r.getReservationId() == id) return r;
+        return null;
+    }
+
+    public List<Reservation> getReservations(){
+        return reservations;
+    }
+
+    /**
+     *
+     * @param room the room where the seats to be checked are placed
+     * @param reservedSeats seats to be checked
+     * @return true if all reserved seats are in one row and there is no gap between them
+     */
+    public boolean canBookSeats(Room room, List<Seat> reservedSeats){
+        reservedSeats.sort(new Comparator<Seat>() {
+            @Override
+            public int compare(Seat o1, Seat o2) {
+                return o1.getColumn() - o2.getColumn();
+            }
+        });
+
+        int rowNum = reservedSeats.get(0).getRow();
+        if(!isSeatValid(room,reservedSeats.get(0))) return false;
+
+        for(int i = 1; i < reservedSeats.size(); i++){
+            if(reservedSeats.get(i).getRow() != rowNum){
+                return false;
+            }
+
+            if(reservedSeats.get(i).getColumn()-1 != reservedSeats.get(i-1).getColumn()){
+                return false;
+            }
+
+            if(!isSeatValid(room,reservedSeats.get(0))) return false;
+
+        }
+        return true;
+    }
+
+    private boolean isSeatValid(Room room, Seat seat){
+        if(room.findSeat(seat) < 0) return false;
+        return !room.isSeatBooked(seat);
+    }
+
+    /**
+     *
+     * @param request object consisting of:
+     * - name
+     * - name
+     * - screening id
+     * - list of reserved seats
+     * @return reservation object and adds it to the list if the entered data is correct. Otherwise, returns null
+     */
+    public Reservation makeReservation( ReservationRequest request) {
```

Comment:
Przydałoby się podzielić tę metodę na mniejsze

Reply:
Tak tak, zgadzam się. Właściwie to nie jestem w stanie na 100% odpowiedzieć dlaczego tego nie zrobiłem :D. Po zaplanowaniu struktury kodu wciąż nie byłem pewien jak będzie wyglądała ta metoda. W planach zakładałem że będzie ona krótsza. Koniec końców w trakcie przygotowania całego rozwiązania funkcja wciąż się rozrastała i tak już zostało